File: frontend/src/services/authService.ts
import axiosInstance from './axiosInstance';
import { setToken, clearToken } from '../utils/tokenUtils';

export const login = async (username: string, password: string) => {
    const response = await axiosInstance.post('/auth/login', { username, password });
    const token = response.data.access_token;
    if (token) {
        setToken(token);
    }
    console.log(response.data);
    return response.data;
};

export const register = async (username: string, password: string) => {
    const response = await axiosInstance.post('/auth/register', {
        username,
        password,
    });

    const token = response.data.access_token;
    console.log(response.data);
    if (token) {
        setToken(token);
    }

    return response.data;
};

export const logout = () => {
    clearToken();
};

export const getMyProfile = async () => {
    const response = await axiosInstance.get('/user/me');
    return response.data;
};

File: Desktop/Services/AuthService.cs
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using Microsoft.Maui.Controls.Platform.Compatibility;
using Microsoft.Maui.Storage;
using System.IdentityModel.Tokens.Jwt;
namespace Desktop.Services;

public static class AuthService
{
    private const string BaseUrl = "http://192.168.100.38:5000";
    private const string TokenKey = "auth_token";
    private static readonly HttpClient client = new();

    public static async Task<string> LoginAsync(string username, string password)
    {
        var payload = new
        {
            username,
            password
        };

        var content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");
        var response = await client.PostAsync($"{BaseUrl}/auth/login", content);

        if(!response.IsSuccessStatusCode)
        {
            throw new Exception("Login failed!");
        }

        var json = await response.Content.ReadAsStringAsync();
        using var doc = JsonDocument.Parse(json);
        var token = doc.RootElement.GetProperty("access_token").GetString();

        if (!string.IsNullOrEmpty(token))
        {
            await SecureStorage.SetAsync(TokenKey, token);
            Preferences.Set(TokenKey, token);
            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
        }

        return json;
    }

    public static string? GetToken()
    {
        return Preferences.ContainsKey(TokenKey) ? Preferences.Get(TokenKey, null) : null;
    }

    public static void Logout()
    {
        SecureStorage.Remove(TokenKey);
        Preferences.Remove(TokenKey);
        client.DefaultRequestHeaders.Authorization = null;
    }

    public static async Task<JsonElement> GetMyProfileAsync()
    {
        var response = await client.GetAsync($"{BaseUrl}/auth/me");

        if (!response.IsSuccessStatusCode)
            throw new Exception("Failed to fetch profile");

        var json = await response.Content.ReadAsStringAsync();
        return JsonDocument.Parse(json).RootElement;
    }

    public static bool IsLoggedIn() => GetToken() != null;

}
